src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -157,7 +157,18 @@ internal sealed class MeterProviderSdk : MeterProvider
                         var metricStreamConfigs = new List<MetricStreamConfiguration>(viewConfigCount);
                         foreach (var viewConfig in this.viewConfigs)
                         {
-                            var metricStreamConfig = viewConfig(instrument);
+                            MetricStreamConfiguration metricStreamConfig = null;
+
+                            try
+                            {
+                                metricStreamConfig = viewConfig(instrument);
+                            }
+                            catch (ArgumentException ex)
+                            {
```

Comment:
`ArgumentException` only captures the ones thrown by the SDK while validating the arguments.. If the user code throws something else, it'd still fallback to "SDK Error Occurred".... Need to comeback to this to see if we should club that into "hey user, your View config is bad, please fix it...."

Reply:
Modified to catch `Exception`